src/decorators.js

```diff
@@ -87,30 +88,66 @@ export function templateController(target?): any {
 */
 export function bindable(nameOrConfigOrTarget?: string | Object, key?, descriptor?): any {
   let deco = function(target, key2, descriptor2) {
-    let actualTarget = key2 ? target.constructor : target; //is it on a property or a class?
+    /**
+     * key2 = truthy => decorated on a class instance
+     * key2 = falsy => decorated on a class
+     */
+    let actualTarget = key2 ? target.constructor : target;
     let r = metadata.getOrCreateOwn(metadata.resource, HtmlBehaviorResource, actualTarget);
     let prop;
+    let propType;
 
     if (key2) { //is it on a property or a class?
       nameOrConfigOrTarget = nameOrConfigOrTarget || {};
       nameOrConfigOrTarget.name = key2;
+      /**
+       * Support for Typescript decorator, with metadata on property type.
+       * Will check for typing only when user didn't explicitly set coerce
+       */
+      if (!nameOrConfigOrTarget.coerce) {
+        propType = metadata.getOwn(metadata.propertyType, target, key2);
+        if (propType) {
+          nameOrConfigOrTarget = classCoerceMap.get(propType) || 'none';
+        }
+      }
```

Comment:
Is this turning on coercion by default in any project that uses TypeScript with emit decorator metadata?

cc @EisenbergEffect

Reply:
I think we need to have a setting that allows people to opt into this. Turning it on automatically, at this point, would be a breaking change to people's apps.